uk.co.compucorp.civicrm.tasksassignments/js/src/tasks-assignments/controllers/task.js

```diff
@@ -3,32 +3,37 @@ define([
     'tasks-assignments/controllers/controllers',
     'tasks-assignments/services/task'
 ], function (moment, controllers) {
-    'use strict';
 
-    controllers.controller('TaskCtrl', ['$scope', '$log', '$rootScope',
-        function ($scope, $log, $rootScope) {
-            $log.debug('Controller: TaskCtrl');
+  'use strict';
 
-            $scope.isCollapsed = true;
-            $scope.picker = { opened: false };
-            $scope.task.activity_date_time = !!$scope.task.activity_date_time ? moment($scope.task.activity_date_time).toDate() : null;
+  controllers.controller('TaskCtrl', ['$scope', '$log', '$rootScope', '$filter',
+    function ($scope, $log, $rootScope, $filter) {
+      $log.debug('Controller: TaskCtrl');
 
-            $scope.dpOpen = function ($event) {
-                $event.preventDefault();
-                $event.stopPropagation();
-                $scope.picker.opened = true;
-            };
+      $scope.isCollapsed = true;
+      $scope.picker = { opened: false };
+      $scope.task.activity_date_time = !!$scope.task.activity_date_time ? moment($scope.task.activity_date_time).toDate() : null;
 
-            $scope.$watch('task.status_id', function (statusId) {
-                var isResolved = $rootScope.cache.taskStatusResolve.indexOf(statusId) > -1;
-                $scope.task.resolved = isResolved;
-                $scope.task.completed = isResolved;
-            });
 
-            $scope.$watch('task.activity_date_time', function (taskDateTime) {
-                $scope.task.due = new Date(taskDateTime).setHours(0, 0, 0, 0) < new Date().setHours(0, 0, 0, 0);
-            });
+      $scope.dpOpen = function ($event) {
+        $event.preventDefault();
+        $event.stopPropagation();
+        $scope.picker.opened = true;
+      };
 
+      $scope.$watch('task.status_id', function (statusId) {
+        var isResolved = $rootScope.cache.taskStatusResolve.indexOf(statusId) > -1;
+        $scope.task.resolved = isResolved;
+        $scope.task.completed = isResolved;
+      });
+
+      $scope.$watch('task.activity_date_time', function (taskDateTime, oldDateTime) {
+        $scope.task.due = new Date(taskDateTime).setHours(0, 0, 0, 0) < new Date().setHours(0, 0, 0, 0);
```

Comment:
UI cannot be the alternate of unit test's. Now you know that its working, but what if somebody changes the condition unknowingly, then how would we know that he made a mistake. Unit tests are there to prevent just that.